tests/CompareTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Test\Semver;
+
+use Composer\Semver\Compare;
+
+class CompareTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @covers Compare::compare
+     * @dataProvider versionComparisons
+     */
+    public function testCompare($v1, $c, $v2, $expected)
+    {
+        if (is_string($expected)) {
+            $this->setExpectedException($expected);
```

Comment:
I think having a separate test for the case of exceptions would be more readable